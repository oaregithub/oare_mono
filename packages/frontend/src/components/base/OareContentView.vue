<template>
  <v-progress-linear v-if="loading" indeterminate />
  <div v-else>
    <slot name="header"></slot>

    <v-card-title
      class="oare-header d-flex justify-space-between align-center test-content-title"
    >
      <slot name="title">
        <div class="d-flex align-center">
          <slot name="title:pre"></slot>
          <span
            class="font-weight-bold"
            v-if="title !== null && title !== undefined"
            >{{ title }}</span
          >
          <slot name="title:post"></slot>
        </div>
      </slot>
    </v-card-title>
    <v-card-text class="black--text">
      <slot></slot>
    </v-card-text>
    <v-card-actions>
      <slot name="actions"></slot>
    </v-card-actions>
  </div>
</template>

<script lang="ts">
import { defineComponent } from '@vue/composition-api';

export default defineComponent({
  name: 'OareContentView',
  props: {
    title: {
      type: String,
      default: null,
    },
    loading: {
      type: Boolean,
      default: false,
    },
  },
});
</script>
