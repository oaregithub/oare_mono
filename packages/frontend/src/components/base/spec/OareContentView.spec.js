import Vuetify from "vuetify";
import VueCompositionApi from "@vue/composition-api";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import OareContentView from "@/components/base/OareContentView.vue";

const localVue = createLocalVue();
localVue.use(VueCompositionApi);

describe("OareContentView", () => {
  let wrapper;
  const TITLE = "Content view title";

  beforeEach(() => {
    wrapper = shallowMount(OareContentView, {
      localVue,
      vuetify: new Vuetify(),
      propsData: {
        title: TITLE,
      },
    });
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("shows title", () => {
    expect(wrapper.find("v-card-title-stub").text()).toBe(TITLE);
  });

  it("shows loading bar when given loading prop", () => {
    wrapper = shallowMount(OareContentView, {
      vuetify: new Vuetify(),
      propsData: {
        title: TITLE,
        loading: true,
      },
    });

    expect(wrapper.find("v-progress-linear-stub")).not.toBe(null);
  });
});
