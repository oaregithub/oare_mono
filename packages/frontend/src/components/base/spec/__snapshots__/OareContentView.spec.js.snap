// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OareContentView matches snapshot 1`] = `
<v-card-stub loaderheight="4" tag="div" maxwidth="100%" class="mx-auto pa-md-7 pa-xs-0 min-height">
  <v-card-title-stub datatestid="content-title" data-testid="content-title" class="display-2 font-weight-bold d-flex justify-space-between align-start">
    Content view title
  </v-card-title-stub>
  <v-card-text-stub class="black--text"></v-card-text-stub>
  <v-card-actions-stub></v-card-actions-stub>
</v-card-stub>
`;
